src/sql/workbench/parts/notebook/electron-browser/outputs/notebookMarkdown.ts

```diff
@@ -199,7 +198,7 @@ export class NotebookMarkdownRenderer {
 			// ignore
 		}
 		let originIndependentUrl = /^$|^[a-z][a-z0-9+.-]*:|^[?#]/i;
-		if (base && !originIndependentUrl.test(href) && (!URI.isUri(href) || !this.exists(URI.parse(href)))) {
+		if (base && !originIndependentUrl.test(href) && (!URI.isUri(href) || !(async () => await this.fileService.exists(URI.parse(href))))) {
```

Comment:
This doesn't fix the issue, you're still going to get a Promise from this. Because your code isn't in an async block, you need to use a .then() on it.

You can't make an async function synchronous.

Reply:
isn't await doing the same thing? awaits until the promise is returned?

Reply:
Nope. You've created an anonymous async function that'll run in the next/end of current event loop. Essentially that if statement will just check if a Promise is returned (which will always be true).

Try this in the chrome console:

``` javascript
const anonymousFunction = (async () => false);

const normalFunction2 = () => {
    if((async () => await anonymousFunction())){
        console.log('boop');
    } else {
        console.log('expected outcome?');
    }
};
normalFunction();

```

Reply:
Not quite. Since this returns a promise you're just checking that the promise is truthy. You need to get the value FROM the promise which as Amir said requires using .then since this is in a non-async function (you can't await)